src/Analyser/MutatingScope.php

```diff
@@ -506,6 +506,13 @@ public function hasVariableType(string $variableName): TrinaryLogic
 	/** @api */
 	public function getVariableType(string $variableName): Type
 	{
+		if (!$this->treatPhpDocTypesAsCertain) {
```

Comment:
The behaviour shouldn't change here - MutatingScope shouldn't be interested in the treat option.

There could be getVariableNativeType method, but I don't think we need it.